presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ActualProperties.java

```diff
@@ -323,7 +322,16 @@ public ActualProperties build()
         {
             List<LocalProperty<VariableReferenceExpression>> localProperties = this.localProperties;
             if (unordered) {
-                localProperties = filteredCopy(this.localProperties, property -> !property.isOrderSensitive());
+                ImmutableList.Builder<LocalProperty<VariableReferenceExpression>> newLocalProperties = ImmutableList.builder();
+                for (LocalProperty<VariableReferenceExpression> property : this.localProperties) {
+                    if (!property.isOrderSensitive()) {
+                        newLocalProperties.add(property);
+                    }
+                    else {
+                        break;
+                    }
```

Comment:
I don't know what these props mean.
Here we are discarding all properties after and including the 1st order sensitive one.
Is this the intent?

Reply:
Yes, this is the intent. It also takes me some time to understand this part of logic, the property is hierarchical and if one is invalid, all following will also be invalid